### DISCUSSION TITLE
How can I make adapters? (for the Virtual Boy and Pokémon Mini)

### DISCUSSION THREAD

**Discussion by ghost** [sanni/cartreader#310] [Need help]
I'm trying to build a Cartreader and thought about the adapters. As I don't know much about Arduinos and building hardware, I found myself wondering how I'd go about designing new adapters, for the Virtual Boy and Pokémon Mini in particular.
I've already read the wiki page about adapters and understand that I'd need to make PCBs for the systems I want to have adapters for and cartridge slots.
What else would I need? Would I have to change the main PCB or something in the build process or software? I know how to program so that shouldn't be a problem, but I don't know what I'd have to do to make things work. I just don't want to make any mistakes during the build process.

**Comment by sanni** [comment]
The Pokemon Mini cart slot looks a lot like the GBA slot so I would start with the GBA adapter as a base but edit the SMD pads size and spacing in Kicad so that they fit the Mini slot. Then I would adapt the pinout according to [here](https://wiki.sublab.net/index.php/PM_Pinouts).
The main PCB doesn't need to be changed.
The dumping software you would need to write from scratch but you can use the already present systems as a guide.

**Reply by ghost** [reply]
Thanks! That really did help. By the way, what would you recommend as the starting point for the Virtual Boy? My guess would be the standard Game Boy.

**Comment by sanni** [comment]
https://mellottsvrpage.com/index.php/vb-cart-programmer-quick-start-guide/ seems to use my Flashrom Writer 16bit functions out of flash.ino to dump Virtual Boy. So hardware-wise I would start by wiring up the Virtual Boy cart slot like the 16bit flash pinout in [pinout.ods](https://github.com/sanni/cartreader/blob/master/pinout.ods) combined with the Virtual Boy pinout from [here](https://www.projectvb.com/vb/tech/cartpinout.html), probably best to start in Kicad with the SNES adapter for that.
And for the software I would start with the existing 29F1610 flashrom functions and see what needs changing there.
